### ISSUE TITLE
Progressive Rollout

### ISSUE THREAD

**Issue by cliveseldon** [SeldonIO/seldon-core#1805]
Look at integration of progressive rollouts. There are two projects focusing on this:

   * [Argo Rollouts](https://argoproj.github.io/argo-rollouts/)
   * [Flagger](https://github.com/weaveworks/flagger)

Look at how one or either could be integrated for ML focused progressive rollouts.

**Comment by anggao** [comment]
We are interest to discuss and contribute to this feature. Flagger will modify user provided resources, it's kind of intrusive and will control all your service resources feel it might not a good fit for Seldon. With Argo-rollouts looks like just need to replace the deployment resource with Rollout CR, then expose the strategy part from Seldon CRD, the integration seems more straightforward.

**Comment by cliveseldon** [comment]
I think we should look at https://github.com/iter8-tools as well

**Comment by anggao** [comment]
@cliveseldon Yes we looked at it as well, it seems also need to manage/create virtualservice and destination rules for you, and there is no manual promotion at the end of the experiment. The only supported mesh solution is Istio, no support for SMI yet.
If adding progressive rollout is on your roadmap for 1.5+ maybe we can discuss this in a meeting or our community call 😄

**Comment by cliveseldon** [comment]
@anggao Yes lets discuss. I know the iter8 team are including CRD support so I think this addresses some of the issues you mention.

**Comment by anggao** [comment]
Great!
